content/blog/fs-advent-2021-fantomas-daemon/index.md

```diff
@@ -0,0 +1,100 @@
+---
+path: '2021/12/07/fs-advent-2021-fantomas-daemon/'
+date: '2021-12-07'
+title: 'Fantomas Daemon'
+tags: ['open-source', 'fsharp', 'fantomas', 'tooling' ]
+cover: './nojaf.com.fantomas-daemon-fsadvent-2021.jpg'
+---
+
+## Intro
+
+The F# advent calendar is a wonderful initiative that inspires the community to create new content in a short period of time.
+I've been fortunate enough to be part of this for some years now. This year, I had in mind to sit this one out actually.
+Sometimes, there are just no new tales to tell and it is better suited to give others an opportunity.
+When I noticed that [Sergey still had some spots left to fill](https://twitter.com/sergey_tihon/status/1459794038115155969?s=20), I did happily volunteer, thought didn't think it through that much.
+
+My original thoughts for this post, was to create some content around the F# compiler. 
+To the benefits of the Fantomas project, I have been contributing [some PR's to the compiler](https://github.com/dotnet/fsharp/pulls?q=is%3Apr+author%3Anojaf) and I wanted to elaborate a bit how a change there can simplify Fantomas.
+This idea for a blogpost sounds reasonable, but life happens and I don't have this content ready at the time of writing.
+So, instead I'd like to write about some of the recent Fantomas changes. Partly because I know this stuff quite well, and partly because it something that I'm proud of.
+
+## Pre-Daemon mode
+
+The next version of [Fantomas 4.6](https://github.com/fsprojects/fantomas/tree/4.6) will contain an improvement in IDE integration.
+Versions 4.5 and prior were available in Ionide, Rider and the Visual Studio extensions.
+Installing those plugins or IDE was sufficient to format code using a shortcut or action.
+Doing this, would call the format function that is present in the [Fantomas NuGet package](https://www.nuget.org/packages/Fantomas/).
+Not only found this format the current file, it would also sync up with the rest of the editor intellisense.
+*And that really is a relevant integration aspect* that you would not get when you format the code using the commandline.
+
+The **downside** of this optimal **integration** is that you are **not in control of the version that is being used in the editor**.
+Fantomas bugfixes almost happen on a weekly basis. I encourage people to scratch their own itch 
```

Comment:
reverse words to "happen almost"; otherwise the meaning is that they almost happen but not completely happen, versus happen almost on a weekly basis, but not always on a weekly basis.